ms-engine-spring-jaxrs/src/main/java/net/trajano/ms/engine/jaxrs/JaxRsPath.java

```diff
@@ -0,0 +1,170 @@
+package net.trajano.ms.engine.jaxrs;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import io.vertx.core.http.HttpMethod;
+
+public class JaxRsPath implements
+    Comparable<JaxRsPath> {
+
+    /**
+     * Mime types that get consumed by the path. This is expected to be empty for
+     * GET requests.
+     */
+    private final String[] consumes;
+
+    private final boolean exact;
+
+    /**
+     * VertX HTTP method.
+     */
+    private final HttpMethod method;
+
+    private final String path;
+
+    private final String pathRegex;
+
+    private final String[] produces;
+
+    public JaxRsPath(final String path,
+        final String[] consumes,
+        final String[] produces,
+        final HttpMethod method) {
+
+        this.path = path;
+        this.consumes = consumes;
+        this.produces = produces;
+        this.method = method;
+        final Pattern placeholderPattern = Pattern.compile("/\\{([^}]+)}");
+        final Pattern regexPlaceholderPattern = Pattern.compile("[-A-Za-z_0-9]+:\\s*(.+)");
+        final Matcher matcher = placeholderPattern.matcher(path);
+
+        final StringBuffer b = new StringBuffer();
+        while (matcher.find()) {
+            final Matcher m2 = regexPlaceholderPattern.matcher(matcher.group(1));
+            if (m2.matches()) {
+                matcher.appendReplacement(b, "/" + m2.group(1));
+            } else {
+                matcher.appendReplacement(b, "/[^/]+");
+            }
+        }
+        matcher.appendTail(b);
+        pathRegex = b.toString();
+        exact = pathRegex.equals(path);
+    }
+
+    /**
+     * Compares two JaxRsPath objects such that it is ordered by most specific and
+     * lowest level first. It sorts it in reverse by path. {@inheritDoc}
+     */
+    @Override
+    public int compareTo(final JaxRsPath o) {
+
+        if (exact && !o.exact) {
+            return -1;
+        } else if (!exact && o.exact) {
+            return 1;
+        }
+
+        return o.path.compareTo(path);
+    }
+
+    @Override
+    public boolean equals(final Object obj) {
+
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        final JaxRsPath other = (JaxRsPath) obj;
+        if (!Arrays.equals(consumes, other.consumes)) {
+            return false;
+        }
+        if (method != other.method) {
+            return false;
+        }
+        if (!Objects.equals(path, other.path)) {
+            return false;
+        }
+        if (!Arrays.equals(produces, other.produces)) {
+            return false;
+        }
+        return true;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1126)